File: package/server/app/schemas/guess_city.py
from pydantic import BaseModel
from typing import List, Optional

class CityCoordinate(BaseModel):
    city: str
    latitude: float
    longitude: float

class GuessRequest(BaseModel):
    photo_id: str
    guess_city: str

class GuessResult(BaseModel):
    correct: bool
    actual_city: str
    distance_km: float
    bearing: float
    direction: str

File: package/website/src/api/guessCity.ts
import request from '@/utils/request'

export interface GuessCityPhoto {
  id: string
  photo_time: string | null
}

export interface CityCoordinate {
  city: string
  latitude: number
  longitude: number
}

export interface GuessRequest {
  photo_id: string
  guess_city: string
}

export interface GuessResult {
  correct: boolean
  actual_city: string
  distance_km: number
  bearing: number
  direction: string
}

export const guessCityApi = {
  /**
   * Get a random photo for the game
   */
  getRandomPhoto() {
    return request.get<GuessCityPhoto>('/api/guess-city/random')
  },

  /**
   * Get all available cities
   */
  getCities() {
    return request.get<CityCoordinate[]>('/api/guess-city/cities')
  },

  /**
   * Verify the guessed city
   */
  guessCity(data: GuessRequest) {
    return request.post<GuessResult>('/api/guess-city/guess', data)
  }
}
